src/pages/ActivityPage/ActivityPage.tsx

```diff
@@ -0,0 +1,173 @@
+    setIsUpdate(true);
+  };
+
+  const handleFilterChange = (value: string) => {
+    setCurrentFilter(value);
+  };
+
+  const handlePageChange = (value: number) => {
+    setCurrentPage(value);
+  };
+
+  return (
+    <Container className={style.profile}>
+      <PageTitle
+        className={style.title}
+      >{`Nice to see you, ${username || 'User'}!`}</PageTitle>
+      <p className={style.sectionTitle}>Your activity</p>
+      <ActivityList userActivity={userActivity} />
+      <p className={style.sectionTitle}>Added roadmaps</p>
+      <div className={style.roadmapListWrapper}>
+        <FilterList filters={FILTER} currentFilter={currentFilter} onChangeFilter={handleFilterChange}/>
+          <ul className={style.roadmapList}>
+            {roadmapList.map(
+              ({ roadmap, completed_vertex_count, vertex_count }) => (
+                <li key={roadmap.title} className={style.listItem}>
+                  <div className={style.itemContent}>
+                    <div className={style.itemHeader}>
+                      <div>
+                        <p className={style.itemHeaderTitle}>{roadmap.title}</p>
+                        <p className={style.itemHeaderInfo}>
+                          {completed_vertex_count? `${completed_vertex_count} of ${vertex_count} topics
+                          completed`: `${vertex_count} topics to do`}
+                        </p>
+                      </div>
+                      <Link to={`${ROUTES.ROADMAPS}/${roadmap.pk}`}>
+                        <Icon
+                          id="icon-open"
+                          stroke="var(--primary-dark)"
+                          fill="var(--white)"
+                          size={28}
+                          boxStyles={style.pointer}
+                        />
+                      </Link>
+                    </div>
+                    <div className={style.progressWrapper}>
+                      <div
+                        className={style.progress}
+                        style={{
+                          width: `${(completed_vertex_count / vertex_count) * 100 + '%'}`,
+                        }}
+                      ></div>
+                    </div>
+                    <ul className={style.tagsList}>
+                      {roadmap.tags.slice(0, 5).map((el) => (
+                        <li key={el} className={style.tags}>
+                          {el}
+                        </li>
+                      ))}
+                    </ul>
+                  </div>
+                  <Button
+                    type="button"
+                    variant="link"
+                    icon={<Icon id="trash" size={20} fill="var(--red)" />}
+                    iconPosition="before"
+                    onClick={() => handleDeleteRoadmap(id, roadmap.pk)}
+                    className={style.trashButton}
+                  >
+                    Delete roadmap
+                  </Button>
+                </li>
```

Comment:
This is a too large part of component, i think you should to bring it into separate file